### PR TITLE
:book: added link to plugin design phase 2 docs

### PR DESCRIPTION
**Pull request by parul5sahoo**
Uodated the plugins page on Kubebuilder book to add latest plugin design phase 2 docs.

### CODE REVIEW

File: docs/book/src/plugins/plugins.md
Comment (camilamacedo86) [inline docs/book/src/plugins/plugins.md:22]: The phase 2 is not here because it is not implemented yet. It is the section vision for plugins see: https://book.kubebuilder.io/plugins/creating-plugins.html#future-vision-for-kubebuilder-plugins

Comment (parul5sahoo) [reply docs/book/src/plugins/plugins.md:22]: oh ohkay so while creating a new plugin, on will have to refer the phase 1 & 1.5?

Comment (parul5sahoo) [reply docs/book/src/plugins/plugins.md:22]: oh ohkay so while creating a new plugin, one will have to refer the phase 1 & 1.5?

Comment (camilamacedo86) [reply docs/book/src/plugins/plugins.md:22]: Yes, I think if you add phase 2 here it might lead to the idea that is implemented right? 

we can add something here as a note with a link to suggest to authors check what is its vision to the future

Comment (parul5sahoo) [reply docs/book/src/plugins/plugins.md:22]: okay.

Comment (parul5sahoo) [reply docs/book/src/plugins/plugins.md]: Hey @camilamacedo86, I made the changes can you take a look?

File: docs/book/src/plugins/plugins.md
Comment (camilamacedo86) [inline docs/book/src/plugins/plugins.md:22]:
```suggestion
You can check the existing design proposal docs at [Extensible CLI and Scaffolding Plugins: phase 1][plugins-phase1-design-doc] and it's [phase 1.5][plugins-phase1-design-doc-1.5] to know more about what is provided by Kubebuilder CLI and API currently. 
```

That is not the only material in the future vision subsection. So, I think the best here is do a new note to just clarifies that.

File: docs/book/src/plugins/plugins.md
Comment (camilamacedo86) [inline docs/book/src/plugins/plugins.md:28]:
```suggestion
<aside class="note">
<h1>What about came next?</h1>
To know more about its plans and the Kubebuilder future vision over the Plugins architecture see the section [Future vision for Kubebuilder Plugins][section-future-vision-plugins].
<.aside>

```

**Comment by parul5sahoo** [comment]
Incorporated the changes..

File: docs/book/src/plugins/plugins.md
Comment (varshaprasad96) [inline docs/book/src/plugins/plugins.md:22]: Nit
```suggestion
You can check the existing design proposal docs at [Extensible CLI and Scaffolding Plugins: phase 1][plugins-phase1-design-doc] and it's [phase 1.5][plugins-phase1-design-doc-1.5] to know more on what is provided by Kubebuilder CLI and API currently. 
```

File: docs/book/src/plugins/plugins.md
Comment (varshaprasad96) [inline docs/book/src/plugins/plugins.md:37]: Nit:
```suggestion
To know more about its plans and the Kubebuilder's future vision over the Plugins architecture see the section [Future vision for Kubebuilder Plugins][section-future-vision-plugins].
```

**Review by varshaprasad96 — APPROVED** [review]
Just nit, else /lgtm from my end.

File: docs/book/src/plugins/plugins.md
Comment (rashmigottipati) [inline docs/book/src/plugins/plugins.md:22]:
```suggestion
You can check the existing design proposal docs at [Extensible CLI and Scaffolding Plugins: phase 1][plugins-phase1-design-doc] and [Extensible CLI and Scaffolding Plugins: phase 1.5][plugins-phase1-design-doc-1.5] to know more about what is provided by Kubebuilder CLI and API currently. 
```

File: docs/book/src/plugins/plugins.md
Comment (rashmigottipati) [inline docs/book/src/plugins/plugins.md:37]:
```suggestion
To know more about Kubebuilder's future vision of the Plugins architecture see [Future vision for Kubebuilder Plugins][section-future-vision-plugins].
```

**Review by rashmigottipati — APPROVED** [review]
/lgtm after addressing all suggestions (or _nits_)

**Comment by parul5sahoo** [comment]
thank you for reviewing it. I've included the changes..

File: docs/book/src/plugins/plugins.md
Comment (camilamacedo86) [inline docs/book/src/plugins/plugins.md:31]: The link alias ought to be always at the bottom of the page.
Could we ensure that before getting this one merged?

**Review by camilamacedo86 — APPROVED** [review]
/lgtm
/approved

**Pull request merged**
